actors/miner/src/lib.rs

```diff
@@ -977,188 +987,213 @@ impl Actor {
+                                "invalid proof for sector {} while requiring activation success: {}",
+                                usi.sector_info.sector_number,
+                                e
+                            ));
+                        }
+                    }
+                }
             }
+        } else {
+            return Err(actor_error!(
+                illegal_argument,
+                "aggregate update proofs not yet supported"
+            ));
+        }
+        if proven_updates.is_empty() {
+            return Err(actor_error!(illegal_argument, "no valid updates"));
+        }
 
-            // Overwrite sector infos.
-            sectors.store(new_sectors).map_err(|e| {
-                e.downcast_default(ExitCode::USR_ILLEGAL_STATE, "failed to update sector infos")
-            })?;
+        // Activate data and compute CommD.
+        let data_activation_inputs: Vec<SectorPiecesActivationInput> = proven_updates
+            .iter()
+            .map(|(update, info)| SectorPiecesActivationInput {
+                piece_manifests: update.pieces.clone(),
+                sector_expiry: info.expiration,
+                sector_number: info.sector_number,
+                sector_type: info.seal_proof,
+                expected_commd: None, // CommD was computed, doesn't need checking.
+            })
+            .collect();
 
-            state.sectors = sectors.amt.flush().map_err(|e| {
-                e.downcast_default(ExitCode::USR_ILLEGAL_STATE, "failed to save sectors")
-            })?;
-            state.save_deadlines(rt.store(), deadlines).map_err(|e| {
-                e.downcast_default(ExitCode::USR_ILLEGAL_STATE, "failed to save deadlines")
-            })?;
+        // Activate data for proven updates.
+        let (batch_return, data_activations) =
+            activate_sectors_pieces(rt, data_activation_inputs, params.require_activation_success)?;
 
-            // Update pledge.
-            let current_balance = rt.current_balance();
-            if pledge_delta.is_positive() {
-                let unlocked_balance =
-                    state.get_unlocked_balance(&current_balance).map_err(|e| {
-                        actor_error!(illegal_state, "failed to calculate unlocked balance: {}", e)
-                    })?;
-                if unlocked_balance < pledge_delta {
-                    return Err(actor_error!(
-                    insufficient_funds,
-                    "insufficient funds for aggregate initial pledge requirement {}, available: {}",
-                    pledge_delta,
-                    unlocked_balance
-                ));
-                }
+        // Successful data activation is required for sector activation.
+        let successful_updates = batch_return.successes(&proven_updates);
+
+        let mut decls_by_deadline = BTreeMap::<u64, Vec<ReplicaUpdateStateInputs>>::new();
+        let mut deadlines_to_load = Vec::<u64>::new();
```

Comment:
You cleaned this up in your notifications PR but can do so here too

Reply:
Done.